### PR TITLE
Custom loader

### PR DESCRIPTION
**Pull request by amrita-syn**
Implements #42 .

### CODE REVIEW

File: src/CornerstoneViewport/CornerstoneViewport.js [hunk 1]
```diff
@@ -79,13 +64,32 @@ class CornerstoneViewport extends Component {
     startLoadHandler: PropTypes.func,
     endLoadHandler: PropTypes.func,
     loadIndicatorDelay: PropTypes.number,
+    loadingIndicatorComponent: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.func,
+    ]),
```
Comment (dannyrb) [inline src/CornerstoneViewport/CornerstoneViewport.js]: I think we can use `PropTypes.element` here?
https://www.npmjs.com/package/prop-types

File: src/CornerstoneViewport/CornerstoneViewport.js [hunk 2]
```diff
@@ -388,9 +403,14 @@ class CornerstoneViewport extends Component {
...
     );
 
-    // Update our "Images Loaded" count.
-    // Better than nothing?
-    this.element[addOrRemoveEventListener](
+    // Update image load progress
+    cornerstone.events[addOrRemoveEventListener](
+      'cornerstoneimageloadprogress',
+      this.onImageProgress
+    );
```
Comment (dannyrb) [inline src/CornerstoneViewport/CornerstoneViewport.js]: I'm pretty sure this is the load progress for each image XHR request. At least in `cornerstone-wado-image-loader`. If that's true, we probably want `onImageLoadProgress` to only update `imageProgress` if it's for the `imageId` at the index we're waiting on.

I don't think this is a blocker, but it's food for thought and probably worth a follow-up enhancement label if you don't want to tackle it this go-round.

File: src/CornerstoneViewport/CornerstoneViewport.js (hunk 1 above)
Comment (amrita-syn) [reply src/CornerstoneViewport/CornerstoneViewport.js]: Just element kept throwing an error in the console, so, made it one of element or function.

File: src/CornerstoneViewport/CornerstoneViewport.js (hunk 2 above)
Comment (amrita-syn) [reply src/CornerstoneViewport/CornerstoneViewport.js]: I think it already does that. I have not verified it though. Also, you are right about not having enough bandwidth right now.

**Pull request merged**